java/src/jmri/jmrix/can/CanFrame.java

```diff
@@ -17,14 +17,83 @@
 @javax.annotation.concurrent.Immutable
 public interface CanFrame {
 
+    /**
+     * Get if the CAN Frame has an extended header
+     * @return true if extended, else false
+     */
     public int getHeader();
 
+    /**
+     * Get if the CAN Frame has an extended header
+     * @return true if extended, else false
+     */
     public boolean isExtended();
 
+    /**
+     * Get if the CAN Frame is an RTR Frame
+     * @return true if RTR, else false
+     */
     public boolean isRtr();
 
+    /**
+     * Get number of data bytes in the frame
+     * @return 0-8
+     */
     public int getNumDataElements();
 
+    /**
+     * Get a single data byte in the frame
+     * @param n the index, 0-7
+     * @return the data element value
+     */
     public int getElement(int n);
 
+    /**
+     * Get formatted monitor String
+     * @return eg. "(12) 81 02 83"
+     */
+    default String monString() {
+        StringBuffer buf = new StringBuffer("(");
+        buf.append( Integer.toHexString(getHeader()));
+        buf.append(isExtended() ? " ext)" : ")");
+        for (int i = 0; i < getNumDataElements(); i++) {
+            buf.append(" " + jmri.util.StringUtil.twoHexFromInt(getElement(i)));
+        }
+        return buf.toString();
+    }
+    
+    
+    /**
+     * Get formatted toString
+     * @return eg. "[12] 81 02 83"
+     */
+    default String getToString() {
+        String s = String.format("[%x] ", getHeader());
```

Comment:
The creation of the returned String in this method should be re-implemented using a StringBuilder to improve performance, especially if called often.

Reply:
Moved to StringBuilder